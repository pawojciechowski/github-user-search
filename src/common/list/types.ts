export interface ListItem {
  id: string;
  text: string;
  href?: string;
}
