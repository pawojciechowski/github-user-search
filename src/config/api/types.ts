// TODO: Verify types

export interface GithubUserResponse {
  login: string;
  name: string | null;
  bio: string | null;
  avatar_url: string;
}

export interface GithubReposResponseRepository {
  id: number;
  name: string;
  html_url: string;
  stargazers_count: number;
}

export interface GithubReposResponse extends Array<GithubReposResponseRepository> {}
