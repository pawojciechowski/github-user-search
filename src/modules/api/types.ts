export type APIUrl = string;

export interface IAPIService {
  get<T>(url: APIUrl, params?: { [key: string]: string }): Promise<T>;
}

export interface APIServiceOptions {
  headers?: { [key: string]: string };
  mockResponse?: { [key: string]: any };
  logRequests?: boolean;
}
