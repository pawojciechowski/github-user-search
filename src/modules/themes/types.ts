import { themeNames } from './themes';

export type Color = string;
export type FontSize = string;
export type FontWeight = string;
export type FontFamily = string;
export type LineHeight = string;
export type LetterSpacing = string;

interface FontSettings {
  fontFamily?: FontFamily,
  fontSize?: FontSize,
  fontWeight?: FontWeight,
  lineHeight?: LineHeight,
  letterSpacing?: LetterSpacing,
  color?: Color
}

interface InputStyles {
  placeholder?: {
    text: FontSettings
  },
  background: Color,
  text: FontSettings
}

interface ButtonStyles {
  background: Color,
  text: FontSettings
}

interface TypographySettings {
  base: FontSettings,
  title: FontSettings
}

export interface Theme {
  background: Color,
  input: InputStyles,
  button: ButtonStyles,
  spacer: number,
  typography: TypographySettings,
}

export type ThemeName = typeof themeNames[number];
