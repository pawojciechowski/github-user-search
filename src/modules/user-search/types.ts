export interface GithubUserResponse {
  login: string,
  name: string,
  bio: string | null,
  avatar_url: string
}

export interface GithubReposResponseRepo {
  id: number,
  name: string,
  html_url: string,
  stargazers_count: number
}

export interface GithubReposResponse extends Array<GithubReposResponseRepo> {}
