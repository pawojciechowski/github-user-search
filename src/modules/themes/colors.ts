const white = '#FFF';
const black = 'rgba(0, 0, 0, 0.87)';
const blue1 = '#452CDC';
const blue2 = '#2F80ED';
const gray1 = '#F0F3F4';
const gray2 = '#E5E5E5';
const gray3 = '#9E9E9E';
const gray4 = '#828282';
const gray5 = '#4F4F4F';

export default {
  white,
  black,
  blue1,
  blue2,
  gray1,
  gray2,
  gray3,
  gray4,
  gray5
};
