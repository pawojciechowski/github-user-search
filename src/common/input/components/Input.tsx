import React from 'react';
import styled, { css } from 'styled-components';
import { Theme } from 'modules/themes/types';
import { baseFontCSS } from 'modules/themes/styles';

const BaseInput = styled.input`
  ${baseFontCSS}
  color: ${({ theme }: { theme: Theme }) => theme.input.text.color};
  line-height: 16px;
  padding: ${({ theme }: { theme: Theme }) => {
    const val = `${theme.spacer * 0.625}px`;
    return `${val} ${val} ${val} 32px`;
  }};
  background: ${({ theme }: { theme: Theme }) => theme.input.background};
  border-radius: 8px;
  border: none;
  box-sizing: border-box;
  width: 100%;

  &::placeholder {
    color: ${({ theme }: { theme: Theme }) => theme.input.placeholder?.text.color};
  }
`;

const InputContainer = ({ className, ...rest }: any) => {
  return (
    <div className={className}>
      <BaseInput {...rest} />
    </div>
  );
};

const Input = styled(InputContainer)`
  position: relative;

  ${({ theme }: { theme: Theme }) => {
    if (theme.input.icon) {
      return css`
        &:before {
          content: '';
          position: absolute;
          left: 8px;
          top: 10px;
          width: 16px;
          height: 16px;
          background-image: url(${theme.input.icon});
        }
      `;
    }
  }}
`;

export default Input;
