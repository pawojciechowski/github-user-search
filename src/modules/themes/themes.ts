import { Theme, ThemeName } from './types';
import colors from './colors';
import searchIcon from 'assets/icons/search-icon.svg';

export const themeNames = ['default'] as const;

export const themes: { [key in ThemeName]: Theme } = {
  default: {
    foreground: colors.white,
    background: colors.gray2,
    secondary: colors.blue2,
    input: {
      placeholder: {
        text: {
          color: colors.gray3,
        },
      },
      text: {
        color: colors.gray5,
      },
      background: colors.gray1,
      icon: searchIcon,
    },
    button: {
      text: {
        color: colors.white,
      },
      background: colors.blue1,
      disabled: {
        text: {
          color: colors.gray4,
        },
        background: colors.gray1,
      },
    },
    spacer: 16,
    typography: {
      base: {
        fontFamily: 'Roboto, sans-serif',
        fontSize: '14px',
        lineHeight: '19px',
        fontWeight: 'normal',
        letterSpacing: '0.4px',
        color: colors.gray4,
      },
      title: {
        fontSize: '18px',
        fontWeight: '500',
        lineHeight: '24px',
        letterSpacing: '0.15px',
        color: colors.black,
      },
    },
    containerShadow: '0px 2px 7px rgba(0, 0, 0, 0.1)',
  },
};
