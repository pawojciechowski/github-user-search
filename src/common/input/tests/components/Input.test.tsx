import React from 'react';
import { render } from '@testing-library/react';
import Input from '../../components/Input';
import { Theme } from 'modules/themes/types';

const testTheme = {
  typography: {
    base: {
      fontSize: 'test',
      fontWeight: 'test',
      lineHeight: 'test',
      letterSpacing: 'test',
      color: 'test',
    },
  },
  spacer: 1,
  input: {
    background: 'test',
    text: {
      color: 'test',
    },
    placeholder: {
      text: {
        color: 'test',
      },
    },
    icon: 'test',
  },
} as Theme;

describe('Input', () => {
  it('matches snapshot', () => {
    const { container } = render(<Input theme={testTheme} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
