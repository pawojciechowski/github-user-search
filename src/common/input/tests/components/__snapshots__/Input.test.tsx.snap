// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
.c1 {
  font-size: test;
  line-height: test;
  font-weight: test;
  -webkit-letter-spacing: test;
  -moz-letter-spacing: test;
  -ms-letter-spacing: test;
  letter-spacing: test;
  color: test;
  color: test;
  line-height: 16px;
  padding: 0.625px 0.625px 0.625px 32px;
  background: test;
  border-radius: 8px;
  border: none;
  box-sizing: border-box;
  width: 100%;
}

.c1::-webkit-input-placeholder {
  color: test;
}

.c1::-moz-placeholder {
  color: test;
}

.c1:-ms-input-placeholder {
  color: test;
}

.c1::placeholder {
  color: test;
}

.c0 {
  position: relative;
}

.c0:before {
  content: '';
  position: absolute;
  left: 8px;
  top: 10px;
  width: 16px;
  height: 16px;
  background-image: url(test);
}

<div
  class="c0"
>
  <input
    class="c1"
  />
</div>
`;
